js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import { mutate } from "swr";
+import { getAardvarkRouteForCurrentDb } from "../../../utils/arangoClient";
+import { useQueryContext } from "../QueryContextProvider";
+import { QueryType } from "./useFetchUserSavedQueries";
+
+const useReadFile = () => {
+  const onReadFile = ({
+    file,
+    onLoad
+  }: {
+    file?: File | null;
+    onLoad: (data: { [key: string]: any }) => void;
+  }) => {
+    if (!file) {
+      return;
+    }
+    const reader = new FileReader();
+    reader.readAsText(file);
+    reader.onload = async () => {
+      const data = reader.result;
+      if (typeof data !== "string") {
+        window.arangoHelper.arangoError(`Failed to parse imported queries`);
+        return;
+      }
+      try {
+        const parsedData = JSON.parse(data);
+        onLoad(parsedData);
+      } catch (e: any) {
```

Comment:
Not a comment but just sharing my learning 
From what I have understood (_feel free to correct me_) for errors without any type I think `unknown` is efficient to use.. `any` is just an escape hatch for TS to ignore and allow all type operations.

Reply:
when I do unknown, I get red swiggly lines still: 
<img width="410" alt="image" src="https://github.com/arangodb/arangodb/assets/2976363/499c0016-d8b2-4dab-9932-3788aaaf4a0b">